watchtower-plugin/src/main.rs

```diff
@@ -133,10 +133,15 @@ async fn get_registration_receipt(
     plugin: Plugin<Arc<Mutex<WTClient>>>,
     v: serde_json::Value,
 ) -> Result<serde_json::Value, Error> {
-    let tower_id = TowerId::try_from(v).map_err(|x| anyhow!(x))?;
+    let tower_id = TowerId::try_from(v.clone()).map_err(|x| anyhow!(x))?;
+    let subscription_start = v["subscription-start"].as_u64().map(|v| v as u32);
+    let subscription_expiry = v["subscription-expiry"].as_u64().map(|v| v as u32);
+
```

Comment:
This is not right. Given we are now potentially passing multiple params to the function, we cannot crate the `TowerId` as before, given `v` now can contain more data.

What is passed the the function is a json of the input params. Check in `convert.rs` how other commands handle this (you'll need to create a `GetRegistrationReceiptParams` struct and implement the parsing)

Reply:
Oh my bad, will fix this.

Reply:
I have implemented `RegistrationReceiptParams` struct, but it seems when optional arguments are passed, the query fails, I haven't been able to figure out exact issue in code, code seems fine to me. Can you pls have a look